### PR TITLE
Add Jenkins integration

### PR DESCRIPTION
**Pull request by konvulse**
Based on the everything Jenkinsfile

See: https://jenk.mct.io/job/conduce-python-api/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#!groovy
+
+properties([buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '14', numToKeepStr: '10')), [$class: 'ScannerJobProperty', doNotScan: false], [$class: 'RebuildSettings', autoRebuild: false, rebuildDisabled: false], pipelineTriggers([])])
+
+
+node('debian') {
+  try {
+    stage("Plan") {
+      checkout scm
+      sh "git submodule update --init --remote"
```
Comment (mbrennek) [inline Jenkinsfile]: this repo doesn't have any submodules, does it?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    print e
+    if (e.getMessage().contains('script returned exit code 143')) {
+      currentBuild.result = "ABORTED"
+      throw e
+    } else {
+      currentBuild.result = "FAILURE"
+
+      if(!env.BRANCH_NAME.contains("PR-")) {
+        slackSend color: "danger", channel: "#jenkins", message: "Build Failure: <${env.JOB_URL}|${env.JOB_NAME}> <${env.BUILD_URL}|#${env.BUILD_NUMBER}> <${env.CHANGE_URL ?: 'https://github.com/ConduceInc/everything'}|(GitHub)>"
+	slackSend color: "danger", channel: "#eng", message: "Build Failure: <${env.JOB_URL}|${env.JOB_NAME}> <${env.BUILD_URL}|#${env.BUILD_NUMBER}> <${env.CHANGE_URL ?: 'https://github.com/ConduceInc/everything'}|(GitHub)>"
```
Comment (mbrennek) [inline Jenkinsfile]: these two lines link to the everything repo, instead of conduce-python-api

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    if (e.getMessage().contains('No changes')) {
+      currentBuild.result = "ABORTED"
+      throw e
+    }
+    currentBuild.result = "FAILURE"
+
+    // Tell everyone we failed
+    if(!env.BRANCH_NAME.contains("PR-")) {
+      slackSend color: "danger", channel: "#jenkins", message: "Build Failure: <${env.JOB_URL}|${env.JOB_NAME}> <${env.BUILD_URL}|#${env.BUILD_NUMBER}> <${env.CHANGE_URL ?: 'https://github.com/ConduceInc/everything'}|(GitHub)>"
+      slackSend color: "danger", channel: "#eng", message: "Build Failure: <${env.JOB_URL}|${env.JOB_NAME}> <${env.BUILD_URL}|#${env.BUILD_NUMBER}> <${env.CHANGE_URL ?: 'https://github.com/ConduceInc/everything'}|(GitHub)>"
```
Comment (mbrennek) [inline Jenkinsfile]: more `slackSend`s with the wrong url

File: tools/jenkins/common.groovy [hunk 4]
```diff
@@ -0,0 +1,44 @@
+def prepare(build_info) {
+  stage("Prepare"){
+    if(!env.BRANCH_NAME.contains("PR-")) {
+      slackSend channel: "#jenkins", message: "Build Started: <${env.JOB_URL}|${env.JOB_NAME}> <${env.BUILD_URL}|#${env.BUILD_NUMBER}> <${env.CHANGE_URL ?: 'https://github.com/ConduceInc/everything'}|(GitHub)>"
```
Comment (mbrennek) [inline tools/jenkins/common.groovy]: everything repo appears in a slackSend again

File: tools/jenkins/common.groovy [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    sh "virtualenv --python=/usr/bin/python3.4 venv"
+    sh ". venv/bin/activate && pip install -r requirements-dev.txt"
+
+    milestone()
+  }
+}
+
+
+def test_and_build() {
+  withEnv(["GLSL_PATH=${env.PWD}/shared/python/victor/glsl/"]) {
```
Comment (mbrennek) [inline tools/jenkins/common.groovy]: this repo doesn't have a `shared/python/victor/glsl` directory.  Leftover copy/paste?

File: Jenkinsfile (hunk 1 above)
Comment (konvulse) [reply Jenkinsfile]: Thanks, it does not.

**Pull request merged**